src/NServiceBus.Core/Persistence/Learning/SagaPersister/LearningSagaIdGenerator.cs

```diff
@@ -0,0 +1,25 @@
+namespace NServiceBus
+{
+    using System;
+    using Sagas;
+
+    class LearningSagaIdGenerator : ISagaIdGenerator
+    {
+        public Guid Generate(SagaIdGeneratorContext context)
+        {
+            if (context.CorrelationProperty == SagaCorrelationProperty.None)
+            {
+                throw new NotSupportedException("The development saga persister doesn't support custom saga finders.");
```

Comment:
Talked to @danielmarbach and we concluded that since it's impossible to even write a finder for the learning transport since users would have no way of querying the data we'll leave this as is.

This issue will go away as we push finder support into the persisters anyway